### PR TITLE
Fix minor option select issues

### PR DESCRIPTION
**Pull request by andysellick**
Several fixes for the `option select` component.

**Adjust option select height calculation:**

- put filter box append code before height calculation code, so height calculation includes filter box
- simplify height calculation code slightly
- change styles for container and content so padding and margin are less of a problem when calculating, fixes scroll being added when only three items and scroll not necessary bug
- change getVisibleCheckboxes to include the first non-visible checkbox, not the last visible one, needed for when only one checkbox with a really long label bug
- add + 1 to the height calculation to remove scrollbar on some browsers

**Refactor the CSS:**

- use proper BEM conventions for classes
- reduce nesting and improve clarity of CSS

Trello card: https://trello.com/c/NIYEeEwu/485-fix-minor-option-select-issues

### CODE REVIEW

File: app/assets/stylesheets/components/_option-select.scss [hunk 1]
```diff
@@ -176,16 +97,91 @@
...
+}
+
+.js-enabled {
+  .app-c-option-select {
+    padding: 0 0 1px;
+  }
+
+  .js-selected-counter {
+    @include core-14;
+  }
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: This class can be removed if you add `govuk-!-font-size-14` to https://github.com/alphagov/finder-frontend/blob/master/app/assets/javascripts/components/option-select.js#L141

File: app/assets/stylesheets/components/_option-select.scss [hunk 2]
```diff
@@ -176,16 +97,91 @@
...
+  display: none;
+
+  .govuk-form-group {
+    margin-bottom: $gutter-one-third;
+  }
+}
+
+.js-enabled {
+  .app-c-option-select {
+    padding: 0 0 1px;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: why is 1px needed?

**Comment by aliuk2012** [comment]
I'm not finished reviewing but I need to work on something urgent.

File: app/assets/stylesheets/components/_option-select.scss (hunk 2 above)
Comment (andysellick) [reply app/assets/stylesheets/components/_option-select.scss]: I think it overrides the non-JS behaviour, which is to have much more padding for some reason, try toggling JS in your browser to see the difference (this looks like my change but I was just moving it around).

File: app/assets/stylesheets/components/_option-select.scss [hunk 3]
```diff
@@ -157,6 +69,22 @@
...
+
+.app-c-option-select__container-head {
+  padding: 0 5px;
+}
+
+.app-c-option-select__label {
+  @include core-19;
+  position: relative;
+  margin-right: 40px;
+  font-weight: bold;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: Font weight and `@inlude core-19` could be replaced with one line`@include govuk-font(48, $weight: bold);`

File: app/assets/stylesheets/components/_option-select.scss [hunk 4]
```diff
@@ -169,16 +97,87 @@
...
+.app-c-option-select__container {
+  position: relative;
+  height: 200px;
+  background-color: $page-colour;
+  overflow-y: auto;
+  overflow-x: hidden;
+}
+
+.app-c-option-select__container-inner {
+  padding: $gutter-one-quarter;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: Could we use govuk static spacing https://design-system.service.gov.uk/styles/spacing/#using-the-static-spacing-function

File: app/assets/stylesheets/components/_option-select.scss [hunk 5]
```diff
@@ -169,16 +97,87 @@
...
 }
 
-.js-closed {
+.app-c-option-select__filter {
+  display: none;
+
+  .govuk-form-group {
+    margin-bottom: $gutter-one-third;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: Could we use govuk static spacing https://design-system.service.gov.uk/styles/spacing/#using-the-static-spacing-function

File: app/assets/stylesheets/components/_option-select.scss [hunk 6]
```diff
@@ -169,16 +97,87 @@
...
+    position: relative;
+    z-index: 100;
+    border: none;
     display: block;
+    width: 100%;
+    text-align: left;
+    cursor: pointer;
+    margin-bottom: -1px;
+    padding: 5px 8px;
+    background-color: $grey-3;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: replace `$grey-3` with `govuk-colour("grey-3")`

File: app/assets/stylesheets/components/_option-select.scss [hunk 7]
```diff
@@ -169,16 +97,87 @@
...
+  }
+}
+
+.js-enabled {
+  .app-c-option-select {
+    padding: 0 0 1px;
+  }
+
+  .app-c-option-select__container {
+    border: 5px solid $grey-3;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: replace `$grey-3` with  `govuk-colour("grey-2")`

File: app/assets/stylesheets/components/_option-select.scss [hunk 8]
```diff
@@ -169,16 +97,87 @@
...
+    margin-bottom: -1px;
+    padding: 5px 8px;
+    background-color: $grey-3;
+
+    &:hover {
+      background-color: $grey-2;
+    }
+
+    &:hover + .app-c-option-select__container {
+      border-color: $grey-2;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: replace colour with `govuk-colour("grey-2")`

File: app/assets/stylesheets/components/_option-select.scss [hunk 9]
```diff
@@ -169,16 +97,87 @@
...
     display: block;
+    width: 100%;
+    text-align: left;
+    cursor: pointer;
+    margin-bottom: -1px;
+    padding: 5px 8px;
+    background-color: $grey-3;
+
+    &:hover {
+      background-color: $grey-2;
```
Comment (aliuk2012) [inline app/assets/stylesheets/components/_option-select.scss]: replace colour with `govuk-colour("grey-2")`

**Pull request merged**